source/extensions/quic_listeners/quiche/platform/quic_logging_impl.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include <cerrno>
+#include <cstring>
+#include <iostream>
+#include <sstream>
+
+#include "common/common/assert.h"
+#include "common/common/logger.h"
+
+#include "absl/base/optimization.h"
+
+#define QUIC_LOG_IMPL_INTERNAL(condition, logstream)                                               \
+  switch (static_cast<int>(condition))                                                             \
+  case 1:                                                                                          \
+    logstream
+
+#define QUIC_LOGGER() Envoy::Logger::Registry::getLog(Envoy::Logger::Id::misc)
+
+#define QUIC_LOG_IF_IMPL(severity, condition)                                                      \
+  QUIC_LOG_IMPL_INTERNAL((condition) && quic::IsLogLevelEnabled(quic::severity),                   \
+                         quic::QuicLogEmitter(quic::severity).stream())
+
+#define QUIC_LOG_IMPL(severity) QUIC_LOG_IF_IMPL(severity, true)
```

Comment:
(optional) slight preference for defining this directly in terms of QUIC_LOG_IMPL_INTERNAL, even though it's longer textually.

Reply:
QUIC_LOG_IF_IMPL is easier to read and I think it compiles to the same code.

Reply:
Fair enough, SG